e2e/features/ui/context-menu-test.js

```diff
@@ -0,0 +1,106 @@
+const reuseables = require('../../reuseables/skip-tour.js');
+const selectors = require('../../reuseables/selectors.js');
+
+const {
+  contextMenu,
+  contextMenuAddMarker,
+  contextMenuDistance,
+  contextMenuArea,
+  contentMenuChangeUnits,
+  geoMeasurementTooltip,
+  measureBtn,
+  unitOfMeasureToggle,
+} = selectors;
+
+const TIME_LIMIT = 1000;
+
+function openContextMenu(c, startX, startY) {
+  c.moveToElement('#wv-map-geographic', startX, startY);
+  c.mouseButtonClick('right');
+  c.waitForElementVisible(contextMenu);
+}
+
+function createDistanceMeasurement(c, [startX, startY], [endX, endY]) {
+  c.waitForElementVisible('#measurement-alert', TIME_LIMIT);
+  c.moveToElement('#wv-map-geographic', startX, startY)
+    .mouseButtonClick(0);
+  c.pause(200);
+  c.moveToElement('#wv-map-geographic', endX, endY)
+    .mouseButtonClick(0)
+    .mouseButtonClick(0)
+    .pause(100);
+}
+
+module.exports = {
+  before: (c) => {
+    reuseables.loadAndSkipTour(c, TIME_LIMIT);
+  },
+
+  'Verify context menu opens on right-click': (c) => {
+    if (c.options.desiredCapabilities.browserName === 'firefox') {
+      return;
+    }
+    openContextMenu(c, 500, 200);
+    c.expect.element(contextMenu).to.be.present;
+  },
+
+  'Adding a marker after clicking "Add Place Marker" in context menu': (c) => {
+    if (c.options.desiredCapabilities.browserName === 'firefox') {
+      return;
+    }
+    c.moveToElement(contextMenuAddMarker, 10, 10);
+    c.mouseButtonClick('left');
+    c.waitForElementVisible('#marker-pin');
```

Comment:
I think this _should_ fail if the marker doesn't show but, wouldn't be a bad idea to add some `assert` or `expect` statement here.